lib/user-profile-manager/user-profile-manager.js

```diff
@@ -48,8 +49,13 @@ UserProfileManager.prototype.init = function (options) { // eslint-disable-line
 		}
 	}
 
-	this.userProfilesServerUrl = options[USER_PROFILE_SERVER_URL] || vcapServiceCredentials[USER_PROFILE_SERVER_URL];
-	this.oauthServerUrl = options[OAUTH_SERVER_URL] || vcapServiceCredentials[OAUTH_SERVER_URL];
+	if (options[constants.APPID_SERVICE_ENDPOINT] && options[constants.APPID_SERVICE_VERSION] && options[constants.APPID_TENANT_ID]) {
+		this.userProfilesServerUrl = options[constants.APPID_SERVICE_ENDPOINT];
+		this.oauthServerUrl = `${options[constants.APPID_SERVICE_ENDPOINT]}/oauth/v${options[constants.APPID_SERVICE_VERSION]}/${options[constants.APPID_TENANT_ID]}`;
+	} else {
+		this.userProfilesServerUrl = options[USER_PROFILE_SERVER_URL] || vcapServiceCredentials[USER_PROFILE_SERVER_URL];
+		this.oauthServerUrl = options[OAUTH_SERVER_URL] || vcapServiceCredentials[OAUTH_SERVER_URL];
+	}
```

Comment:
Check what are the error messages in case that APPID_SERVICE_ENDPOINT field exists and another field is missing (e.g. version)